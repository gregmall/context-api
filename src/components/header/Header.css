.dark{
  background-color: rgb(17, 26, 51);
  color:rgb(41, 201, 212);
  height: 75px;
  width: 100%;
  margin-bottom: 10px;
  display: flex;
  justify-content: space-between;;
  flex-direction: row;
  

}
.light{
  background-color:rgb(41, 201, 212);
  color: rgb(17, 26, 51);
  height: 75px;
  width: 100%;
  margin-bottom: 10px;
  display: flex;
  justify-content: space-between;;
  flex-direction: row;
}
