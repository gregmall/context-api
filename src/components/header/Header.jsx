import React from 'react';
import { useThemeToggle } from '../../hooks/character';
import { useThemeType } from '../../hooks/character';
import styles from './Header.css'


const Header = () => {
  const toggle = useThemeToggle();
  const type = useThemeType();
  console.log(toggle)
  return (
    <header className={styles[type]}>
  
      <div>
          {/* <img src="" /> */}
      </div>
      <div>
          <a href="/">HOME</a>
      </div>
      <div>
        <label>Toggle for dark or light layout</label>
        <button onClick={toggle}>{type}</button>
        
      </div>
      <div>
          <h1>
            RICK AND MORTY
          </h1>
      </div>
    </header>
  );
};
export default Header;
