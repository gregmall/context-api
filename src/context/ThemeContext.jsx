import React from 'react';
import { useState } from 'react';

export const ThemeContext = React.createContext();

export const ThemeProvider = ({ children }) => {
  const [themeType, setThemeType] = useState('dark');

  const toggle = () => {
    setThemeType(oldTheme => {
      if(oldTheme ==='dark') return 'light';
      return 'dark';
    });
  };

  return (
    <ThemeContext.Provider value ={{ themeType, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};
