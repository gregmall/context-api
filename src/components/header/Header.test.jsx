import React from 'react'
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';
import { ThemeProvider } from '../../context/ThemeContext';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders the Header to the page', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    
    );
      expect(asFragment()).toMatchSnapshot();
  });
});
