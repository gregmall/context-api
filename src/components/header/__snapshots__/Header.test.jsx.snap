// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the Header to the page 1`] = `
<DocumentFragment>
  <header
    class="dark"
  >
    <div>
      <img
        src="src/components/header/Webp.net-resizeimage.png"
      />
    </div>
    <div>
      <a
        href="/"
      >
        HOME
      </a>
    </div>
    <div>
      <label>
        Toggle for dark or light layout
      </label>
      <button
        data-testid="button"
      >
        dark
      </button>
    </div>
    <div>
      <h1>
        RICK AND MORTY
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
